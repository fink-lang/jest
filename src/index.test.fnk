{describe, it, skip, only, expect, any, mock, mock_fn} = import '.'
{to_match_snapshot, to_equal, to_not_equal} = import '.'
{was_called, was_not_called, was_called_with, was_not_called_with} = import '.'
{to_throw, to_not_throw} = import '.'


mock_test = mock_fn fn: 'testing'

mock '@fink/test'
  fn: dict:
    test: mock_test
  {virtual: true}

{test} = import '@fink/test'


only.describe 'test utils', fn:

  only.it 'equality', fn:
    expect
      'foobar'
      to_equal 'foobar'

    expect
      'foobar'
      to_not_equal 'spam'


  only.it 'tests mock funks', fn:
    foo = mock_fn fn: null

    expect
      foo
      was_not_called

    foo 'bar'

    expect
      foo
      was_called

    expect
      foo
      was_called_with any String

    expect
      foo
      was_not_called_with any Number

    expect
      foo
      was_called_with 'bar'


  only.it 'test mock module', fn:
    test 'testing'
    expect
      mock_test
      was_called_with 'testing'


  only.it 'tests exceptions', fn:
    foo = mock_fn
      # istanbul ignore next
      fn msg:
        throw new Error msg
        null

    expect
      fn: foo 'test'
      to_throw 'test'

    expect
      fn: 'test'
      to_not_throw()

  skip.it 'skips',
    # istanbul ignore next
    fn:
      throw new Error 'fail'
      null


skip.describe 'test utils', fn:
  undefined