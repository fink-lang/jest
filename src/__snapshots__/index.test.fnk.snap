// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process compiles fink 1`] = `
Object {
  "code": "const foo = 1234;
Object.assign(module.exports, {
  foo
});",
  "map": Object {
    "mappings": "MAAA,G,GAAM,I",
    "names": Array [],
    "sources": Array [
      "foobar.fnk",
    ],
    "sourcesContent": Array [
      "foo = 1234",
    ],
    "version": 3,
  },
}
`;
